lib/message/encrypt.js

```diff
@@ -15,30 +15,26 @@ export default async function encryptMessage({ armor = true, ...options }) {
     options.date = typeof options.date === 'undefined' ? serverTime() : options.date;
     options.compression = options.compression ? openpgp.enums.compression.zlib : undefined;
 
+    options.armor = armor;
+
     if (options.detached) {
         // Create detached signature of message
         const signature = await options.message.signDetached(options.privateKeys);
 
         // Encrypt message without signing it
-        const result = await openpgp.encrypt({
-            ...options,
-            privateKeys: [],
-            armor
-        });
+        options.privateKeys = [];
+        const result = await openpgp.encrypt(options);
 
         // Encrypt signature and add it to the final result
-        const encryptedSignature = await openpgp.encrypt({
-            ...options,
-            message: createMessage(signature.packets.write()),
-            privateKeys: [],
-            armor
-        });
-        result.encryptedSignature = armor ? encryptedSignature.data : encryptedSignature.message;
+        options.message = createMessage(signature.packets.write());
+        options.sessionKey = result.sessionKey;
```

Comment:
This works if `options.returnSessionKey` is set, but breaks things if `options.sessionKey` is set but `options.returnSessionKey` isn't, I think. You could force `returnSessionKey = true` before encrypting the message above, if you want - I think it shouldn't do any harm. If you want to be super safe about things, you could add another test where you pass `sessionKey` but not `returnSessionKey`.